packages/js-sdk/src/sandbox.ts

```diff
@@ -170,149 +264,253 @@ export class Sandbox extends SandboxBase {
   /**
    * Left click on the current mouse position.
    */
-  async leftClick() {
-    return this.runPyautoguiCode('pyautogui.click()')
+  async leftClick(): Promise<void> {
+    await this.commands.run("xdotool click 1", { envs: { DISPLAY: this.display } });
   }
 
   /**
    * Double left click on the current mouse position.
    */
-  async doubleClick() {
-    return this.runPyautoguiCode('pyautogui.click(clicks=2, interval=0.25)')
+  async doubleClick(): Promise<void> {
+    await this.commands.run("xdotool click --repeat 2 1", { envs: { DISPLAY: this.display } });
   }
 
   /**
    * Right click on the current mouse position.
    */
-  async rightClick() {
-    return this.runPyautoguiCode('pyautogui.rightClick()')
+  async rightClick(): Promise<void> {
+    await this.commands.run("xdotool click 3", { envs: { DISPLAY: this.display } });
   }
 
   /**
    * Middle click on the current mouse position.
    */
-  async middleClick() {
-    return this.runPyautoguiCode('pyautogui.middleClick()')
+  async middleClick(): Promise<void> {
+    await this.commands.run("xdotool click 2", { envs: { DISPLAY: this.display } });
   }
 
   /**
    * Scroll the mouse wheel by the given amount.
+   * @param direction - The direction to scroll. Can be "up" or "down".
    * @param amount - The amount to scroll.
    */
-  async scroll(amount: number) {
-    return this.runPyautoguiCode(`pyautogui.scroll(${amount})`)
+  async scroll(direction: 'up' | 'down' = 'down', amount: number = 1): Promise<void> {
+    const button = direction === 'up' ? '4' : '5';
+    await this.commands.run(
+      `xdotool click --repeat ${amount} ${button}`, { envs: { DISPLAY: this.display } }
+    );
+  }
+
+  /**
+   * Move the mouse to the given coordinates.
+   * @param x - The x coordinate.
+   * @param y - The y coordinate.
+   */
+  async moveMouse(x: number, y: number): Promise<void> {
+    await this.commands.run(
+      `xdotool mousemove --sync ${x} ${y}`, { envs: { DISPLAY: this.display } }
+    );
+  }
+
+  /**
+   * Get the current cursor position.
+   * @returns A object with the x and y coordinates or null if the cursor is not visible.
+   */
+  async getCursorPosition(): Promise<CursorPosition | null> {
+    const result = await this.commands.run(
+      "xdotool getmouselocation", { envs: { DISPLAY: this.display } }
+    );
+
+    const match = result.stdout.match(/x:(\d+)\s+y:(\d+)/);
+    if (match) {
+      const [, x, y] = match;
+      if (x && y) {
+        return { x: parseInt(x), y: parseInt(y) };
+      }
+    }
+    return null;
```

Comment:
I am not sure if returning null is a good idea, either return { x: undefined, y: undefined } or throw?

Reply:
@mlejva Please decide.

Reply:
Throw an Error.